include/gridtools/thread_pool/omp.hpp

```diff
@@ -0,0 +1,50 @@
+/*
+ * GridTools
+ *
+ * Copyright (c) 2014-2019, ETH Zurich
+ * All rights reserved.
+ *
+ * Please, refer to the LICENSE file in the root directory.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#pragma once
+
+#ifdef _OPENMP
+#include <omp.h>
+#endif
+
+namespace gridtools {
+    namespace thread_pool {
+        struct omp {
+#ifdef _OPENMP
+            friend auto thread_pool_get_thread_num(omp) { return omp_get_thread_num(); }
+            friend auto thread_pool_get_max_threads(omp) { return omp_get_max_threads(); }
+
+            template <class F, class I>
+            friend void thread_pool_parallel_for_loop(omp, F const &f, I lim) {
+#pragma omp parallel for collapse(1)
+                for (I i = 0; i < lim; ++i)
+                    f(i);
+            }
+
+            template <class F, class I, class J>
+            friend void thread_pool_parallel_for_loop(omp, F const &f, I i_lim, J j_lim) {
+#pragma omp parallel for collapse(2)
+                for (I i = 0; i < i_lim; ++i)
+                    for (J j = 0; j < j_lim; ++j)
```

Comment:
Not sure if we even have to provide these specializations as the collapse-thing does exactly the same thing as the default implementation anyway (when we make sure that the loop order is the same in the default implementation).

Reply:
I would prefer to keep those specializations. I am not so sure about omp implementation. Maybe it can do better.

Reply:
As far as I remember I once accidentally checked the assembly of ICC on KNL. It used 64bit integers in the division which are quite slow, so I stumbled upon it during the performance analysis using perf… But I am fine with keeping it.